### DISCUSSION TITLE
Deploy Tina app to hyperscalers?

### DISCUSSION THREAD

**Discussion by paolosfilio** [tinacms/tinacms#2348] [Help] [answered]
Is it possible to deploy a TinaCMS app to hyperscalers like, e.g. Firebase or AWS? If yes do you know if any documentation exists how to do so?

**Comment by jeffsee55** [comment, marked as answer]
At the moment, we are focusing on Next.js and leveraging `getStaticProps` as the only way you can fetch data. What this means is that during build time we require that our http server can run on `localhost` from within CI and serve the data for `getStaticProps`. From what we've seen this works great on Netlify and Vercel, but doesn't currently work on AWS. We have plans to remove that requirement but we aren't there yet.

Aside from that, Tina stays out of your way for the most part, so anywhere Next.js can run should be good to go.

**Comment by georgibakken** [comment]
Is there a possibility to deploy a tina app to Azure [static web apps](https://learn.microsoft.com/en-gb/azure/static-web-apps/nextjs)? They have recently released support for nextjs static and SSR. 

I am using the [tina docs starter](https://github.com/tinacms/tina-docs-starter)

**Reply by kldavis4** [reply]
Based on that linked web page, it doesn't look like there is any reason why you couldn't use it. Please let us know how it goes!

**Reply by georgibakken** [reply]
I have tried, and seems to be some limitation with the static web app, as the tina-starter doesn't work well with next export. And for regular site, the total app size is crossing the 100MB limit Azure has set, mostly due to node_modules and the cache.

I see next.js can be [deployed as a docker image](https://nextjs.org/docs/deployment#docker-image), I will try this with Tina

**Reply by georgibakken** [reply]
Hello again.

Actually I managed to deploy the tina to an azure static web app, however it's acting a bit funky.

If I am not logged in via admin, the page does not have any styling. It's missing the styling and tina-tailwind chunk. The deployed page is: https://ashy-mushroom-01c9fd703.2.azurestaticapps.net

If I log in as admin, the page has the styling. Have tried to reproduce it locally with `yarn build` and `yarn start` but there everything seems fine. 

I am assuming it's a azure static web app issue.
